### PR TITLE
Updated display configuration instructions for Bookworm changes

### PR DESCRIPTION
**Pull request by nathan-contino**
Updates the display configuration instructions to use the new `vc4-kms-dpi-generic` overlay instead of the deprecated `dpi_timings` configuration.

Requested `pixel-freq` clarification has been updated in the legacy documentation. But that documentation still refers to `hdmi_timings`, a configuration that's even more outdated than `dpi_timings`. Should we replace this doc with the current `dpi_timings` doc?

Closes https://github.com/raspberrypi/documentation/issues/2989

### CODE REVIEW

**Comment by lurch** [comment]
`hdmi_timings` has already been moved to https://github.com/raspberrypi/documentation/blob/bookworm/documentation/asciidoc/computers/legacy_config_txt/video.adoc

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
+Display Parallel Interface (DPI) displays can be connected to Raspberry Pi devices via the 40-pin general-purpose input/output (GPIO) connector as an alternative to using the dedicated Display Serial Interface (DSI) or High-Definition Multimedia Interface (HDMI) ports. Many third-party DPI displays have been made available to take advantage of this. The Buster (and earlier) Raspberry Pi operating system (OS) and the legacy display stack used Raspberry Pi-specific parameters in `config.txt` to configure DPI displays. With the move to Bullseye and its use of the Kernel Mode Setting (KMS) graphics driver by default, these `config.txt` entries are no longer relevant as all control of the display pipeline has shifted to the Linux kernel.
 
-This whitepaper assumes that the Raspberry Pi is running the Raspberry Pi OS (Linux), and is fully up to date with the latest firmware and kernels.
+This whitepaper assumes that the Raspberry Pi runs Raspberry Pi OS (Linux) and is fully up to date with the latest firmware and kernels.
```
Comment (aallan) [inline documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:7]: This change is going to cause merge conflicts with @jehlers42 #3087 PR.

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc [hunk 2]
```diff
@@ -335,162 +335,115 @@ h|*27* h|*26* h|*25* h|*24* h|*23* h|*22* h|*21* h|*20* h|*19* h|*18* h|*17* h|*
 
 === Disable Other GPIO Peripherals
 
-Note that all other peripheral overlays that use conflicting GPIO pins must be disabled. In config.txt, take care to comment out or invert any dtparams that enable I2C or SPI:
+Note that all other peripheral overlays that use conflicting GPIO pins must be disabled. In `config.txt`, take care to comment out or invert any dtparams that enable I2C or SPI:
```
Comment (aallan) [inline documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:338]: Please don't start a sentence with "Note" unless you're going to make a `NOTE:` block.

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc [hunk 3]
```diff
@@ -335,162 +335,115 @@ h|*27* h|*26* h|*25* h|*24* h|*23* h|*22* h|*21* h|*20* h|*19* h|*18* h|*17* h|*
...
-   1: RGB Data changes on falling edge and is stable at rising edge.
-
-hsync/vsync/output_enable_polarity:
-   0: default for HDMI mode
-   1: inverted
-
-hsync/vsync/oe phases:
-   0: DPI_PHASE_POSEDGE
-   1: DPI_PHASE_NEGEDGE
+`display_auto_detect=1`
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:355]: This only applies to DSI displays, so might not be useful in the DPI docs?

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc [hunk 4]
```diff
@@ -335,162 +335,115 @@ h|*27* h|*26* h|*25* h|*24* h|*23* h|*22* h|*21* h|*20* h|*19* h|*18* h|*17* h|*
...
 ----
 
-NB the single bit fields all act as an "invert default behaviour".
-
-=== Controlling Timings and Resolutions
+Replace the `1` with a `0` to disable auto detect.
+When you connect the official Raspberry Pi display with auto detect enabled, KMS determines the display model automatically and configures the appropriate display settings.
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:359]: "the official Raspberry Pi display" is also a DSI device.

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc [hunk 5]
```diff
@@ -335,162 +335,115 @@ h|*27* h|*26* h|*25* h|*24* h|*23* h|*22* h|*21* h|*20* h|*19* h|*18* h|*17* h|*
...
+==== Manually Configure a Display
 
-The `dpi_group` and `dpi_mode` config.txt parameters are used to set either predetermined modes (DMT or CEA modes as used by HDMI) or a user can generate https://forums.raspberrypi.com/viewtopic.php?f=29&t=24679[custom modes].
+NOTE: In Raspberry Pi OS _Bookworm_ or later, the `dpi_output_format` and `dpi_timings` entries in `config.txt` previously used to set up DPI have been superseded by the `vc4-kms-dpi-generic` overlay.
 
-If you set up a custom DPI mode, then in config.txt use:
+To use any display other than the official Raspberry Pi display, you must specify a `dtoverlay` entry in `config.txt`. The panel manufacturer should configure timings for your display in Linux kernel code and provide an overlay to enable those settings. See the https://github.com/raspberrypi/linux/blob/rpi-5.10.y/arch/arm/boot/dts/overlays/vc4-kms-kippah-7inch-overlay.dts[Adafruit Kippah display entry] for an example. The following example demonstrates how to set a `dtoverlay` entry for the Kippah display in your xref:config_txt.adoc#what-is-config-txt[`/boot/firmware/config.txt`] file:
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:365]: Small nitpick, but could we link to the `rpi-6.1.y` branch here? See also #3076

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc [hunk 6]
```diff
@@ -335,162 +335,115 @@ h|*27* h|*26* h|*25* h|*24* h|*23* h|*22* h|*21* h|*20* h|*19* h|*18* h|*17* h|*
...
+dtoverlay=vc4-kms-dpi-generic,hactive=480,hfp=26,hsync=16,hbp=10
+dtparam=vactive=640,vfp=25,vsync=10,vbp=16
+dtparam=clock-frequency=32000000,rgb666-padhi
 ----
 
-=== Overlays
+NOTE: Device tree line length must not exceed 80 characters. When a setting requires a line longer than 80 characters, split the assignment of that parameter across multiple lines.
 
-A Linux Device Tree overlay is used to switch the GPIO pins into the correct mode (alt function 2). As previously mentioned, the GPU is responsible for driving the DPI display. Hence there is no Linux driver; the overlay simply sets the GPIO alt functions correctly.
+Paramater display tree definitions support the following options:
```
Comment (lurch) [inline documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:384]: Paramater -> Parameter

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc (hunk 3 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:355]: I might be misunderstanding, but I think DSI refers to a connector standard (cable on the Pi, as opposed to plugging in a microHDMI cable) and DPI (display pixel interface) refers to the communication standard, which in Bookworm now includes both HDMI and DSI. I think. I'm not sure it makes sense to split the two apart any more given that change.

Comment (lurch) [reply documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:355]: DSI = Display Serial Interface
DPI = Display Parallel Interface

So different signalling mechanisms, not just different types of connector.

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc (hunk 1 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc]: TY for the heads-up. Reversed this, which should simplify the git shenanigans later.

File: documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc (hunk 3 above)
Comment (aallan) [reply documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:355]: Wait! DPI = Display Pixel Interface surely?

https://www.itwissen.info/en/display-pixel-interface-DSI-DPI-128674.html#gsc.tab=0 (amongst other references?)

Comment (lurch) [reply documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:355]: I just asked @6by9 and he confidently said "Display Parallel Interface".
https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc
https://pinout.xyz/pinout/dpi

I guess it's just an unfortunate coincidence that "DPI" has two different expansions, depending on context :joy: :upside_down_face:

Comment (6by9) [reply documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:355]: And just to make it murkier still, the actual title of the spec is
> MIPI Alliance Standard for Display Pixel Interface (DPI-2)

Whilst the spec is at V2, I wouldn't say that the ```2``` on the end of ```DPI-``` refers to the version, in the same way as we implement CSI-2 v1.01 for cameras.

Comment (aallan) [reply documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc:355]: I hate displays. I just want to make that really clear. 🤦‍♂️

**Pull request merged**